transaction/std/src/memo_builder/gift_code_sender_memo_builder.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Defines the Memo Builder for the gift code sender memo (0x0002)
+//! specified in MCIP #32
+
+use crate::{memo::GiftCodeSenderMemo, MemoBuilder, ReservedDestination};
+use mc_account_keys::PublicAddress;
+use mc_transaction_core::{Amount, MemoContext, MemoPayload, NewMemoError};
+
+/// There are three possible gift code memo types specified in MCIP #32
+/// | Memo type bytes | Name                        |
+/// | -----------     | -----------                 |
+/// | -->0x0002<--    | Gift Code Sender Memo       |
+/// |    0x0201       | Gift Code Funding Memo      |
+/// |    0x0202       | Gift Code Cancellation Memo |
+/// This memo builder builds the gift code sender memo (0x0002). A gift code
+/// considered redeemed when the Receiver uses the TxOut spend private key
+/// of the gift code TxOut they received from the Sender to send the TxOut
+/// from the sender's gift code subaddress to their own change subaddress.
+/// The sender memo is written into the TxOut the receiver sends to the
+/// change address and includes an optional Utf-8 note up to 64 bytes long
+/// the Receiver can  use to record any information they desire about the
+/// gift code.
+#[derive(Clone, Debug)]
+pub struct GiftCodeSenderMemoBuilder {
+    // Utf-8 formatted note within the memo that can be up to 64 bytes long
+    note: String,
+    // Whether we've already written the change memo
+    wrote_change_memo: bool,
+}
+
+impl GiftCodeSenderMemoBuilder {
+    /// Initialize memo builder with a utf-8 note (up to 64 bytes), This
+    /// method will enforce the 64 byte limit with a NewMemoErr if the
+    /// note passed is longer than 64 bytes.
+    pub fn new(note: &str) -> Result<Self, NewMemoError> {
+        if note.len() > GiftCodeSenderMemo::MEMO_DATA_LEN {
+            return Err(NewMemoError::BadInputs(
+                "Sender note cannot be longer than 64 bytes".into(),
```

Comment:
Should use `MEMO_DATA_LEN` instead of 64

Reply:
Noted!